src/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/GitCommit.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Xamarin.Tools.Zip;
+
+using IOFile = System.IO.File;
+
+namespace Xamarin.Android.Tools.BootstrapTasks
+{
+	public sealed class GitCommit : Task
+	{
+		[Required]
+		public  ITaskItem       Directory                   { get; set; }
+
+		[Output]
+		public  string          AbbreviatedCommitHash       { get; set; }
+
+		[Output]
+		public  string          Branch                      { get; set; }
+
+		public override bool Execute ()
+		{
+			Log.LogMessage (MessageImportance.Low, $"Task {nameof (GitCommit)}");
+			Log.LogMessage (MessageImportance.Low, $"  {nameof (Directory)}: {Directory.ItemSpec}");
+
+			AbbreviatedCommitHash   = GetAbbreviatedCommitHash ();
+			Branch                  = GetBranch ();
+
+			Log.LogMessage (MessageImportance.Low, $"  [Output] {nameof (AbbreviatedCommitHash)}: {AbbreviatedCommitHash}");
+			Log.LogMessage (MessageImportance.Low, $"  [Output] {nameof (Branch)}: {Branch}");
+
+			return !Log.HasLoggedErrors;
+		}
+
+		string GetGitOutput (string arguments)
+		{
+			var p   = new Process () {
+				StartInfo = {
+					FileName                = "git",
```

Comment:
Does it need to be `git.exe` on Windows? Or will this approach not work *at all*?

Please elaborate.